### ISSUE TITLE
Empty CSV file in some conditions

### ISSUE THREAD

**Issue by wil70** [eugeneilyin/nrdtocsv#2]
Hello, 

Great addon - TY!

I found an issue, I downloaded one day (for example 2023-10-10) of "ES 12-23" from NT
![image](https://github.com/eugeneilyin/nrdtocsv/assets/7807709/37c403ba-7971-48df-8930-0285b78df76d)

I can see in the Replay data being there 
![image](https://github.com/eugeneilyin/nrdtocsv/assets/7807709/5acf0abb-f32a-4819-93ac-91a43e2f9938)
and also the file downloaded is there
![image](https://github.com/eugeneilyin/nrdtocsv/assets/7807709/d661eda1-f88c-4535-8cd4-34da18377a86)

but when I use the converter the NRD to CSV, it generates an empy CSV file?
![image](https://github.com/eugeneilyin/nrdtocsv/assets/7807709/32ea47f7-7694-4f0d-b56a-95e3ae5c60bc)

![image](https://github.com/eugeneilyin/nrdtocsv/assets/7807709/e9bf3749-8015-42bb-90fd-70d689f0a63a)

I'm using NRDToCSV-1.2.0.zip

Thanks for yoru help

Wil

**Comment by eugeneilyin** [comment]
Seems like the format of the `20231010.nrd` file us not compatible with NT internal conversion functionality or the file is not accessible or file writes are not permitted (less possible because the empty `*.cvs` file was created). Maybe the new updated file format provided which is not compatible with the internal NT conversion utility:
- Try to retrieve and convert some other (external `*.nrd` files) in your current NT version
- Try to move your `*.nrd` file to some short (non spaces) test folder, like `C:\MarketData\20231010.nrd` and convers all (keep filter empty)
- Check folder access permissions
- Try another instrument or use smaller file

In any case the utility is only a useful wrapper around the NT functionality.

**Comment by wil70** [comment]
Thanks @eugeneilyin, this is a great little utility!

Yes I can write in the folder. Same issue with other instruments.

I think this is an date index issue, why? because when I do several files then most of the files are being processed but not all (it seems first and last are not). If I do one file as describe above then it is not processesed. 


**Detail:** Here is what I tried with multiple files
I downloaded a few ES nrd. as you can see they are not empt files.

![image](https://github.com/eugeneilyin/nrdtocsv/assets/7807709/5d367511-8f16-473f-8001-b2fb271293f6)
and
![image](https://github.com/eugeneilyin/nrdtocsv/assets/7807709/4b30ba00-d8fe-47d8-b6cb-5eaf02d2ad8b)

When I use the converter, it find the nrd and process them but most are empty except one csv file
![image](https://github.com/eugeneilyin/nrdtocsv/assets/7807709/a77284b9-8a00-4701-90f2-9b5ba8cca0bb)
![image](https://github.com/eugeneilyin/nrdtocsv/assets/7807709/fc480202-c6c0-4757-89fb-3bf909008cfe)

**Comment by wil70** [comment]
Hello, any update on this?
TY

**Comment by ck99** [comment]
I think I've seen the same issue. 

In my case, everything worked fine when I had NinjaTrader timezone set to the same as my pc (UTC +00:00). Then, I changed the timezone of NinjaTrader to (UTC -06:00) and then things got weird, but, behaved properly once I changed the timezone back to (UTC +00:00)

> I think this is an date index issue, why? because when I do several files then most of the files are being processed but not all (it seems first and last are not). If I do one file as describe above then it is not processesed.

For example, I downloaded replay data for 20251120 (today), and the CSV extraction file came up empty for that date, as @wil70  described. But, I noticed that the csv for 20251119 actually had the data for 20251120 inside it. 

As @eugeneilyin  said, this tool just dumps what NT provides, but, I also think that NT has some strange timezone handling logic (e.g. its suspicious that you need to restart NT after you change the timezone.)

In my case, since I want NT to have (UTC -06:00), I just toggle it to my local timezone before I run nrdtocsv, and then toggle it back afterwards.

**Comment by GatorAli** [comment]
Yes there must be a timezone or maybe date issue. In my case Filename 20250925 - all the entries carry a date of 26 instead of 25. but the data of 25.

The file for 26 is than empty like in @wil70 case, because the 27th (which would be the timestamp in the data) is a saturday.

In the code there is a line which says entry.date.AddDays(1) - i dont know much about C# but seemed oddly fitting to my problem.

I will try the timezone thing. Overall im Happy this converter exists, it helps alot tho tweek things.

**Comment by chesscow** [comment]
I would like to comment that this bug still exists.  Author can we get an update? thank you!

**Comment by chesscow** [comment]
I've run it through CHATGPT and it seems to work now

[NRDToCSV 3rd party fix.zip](https://github.com/user-attachments/files/25996607/NRDToCSV.3rd.party.fix.zip)
